packages/devtools_app/lib/src/screens/inspector_shared/inspector_controls.dart

```diff
@@ -37,19 +40,37 @@ class InspectorControls extends StatelessWidget {
           ),
           builder: (_, selectModeSupported, __) {
             return ServiceExtensionButtonGroup(
+              fillColor: activeButtonColor,
               extensions: [
                 selectModeSupported
                     ? extensions.toggleSelectWidgetMode
                     : extensions.toggleOnDeviceWidgetInspector,
               ],
               minScreenWidthForTextBeforeScaling:
-                  InspectorScreenBodyState.minScreenWidthForTextBeforeScaling,
+                  InspectorScreen.minScreenWidthForTextBeforeScaling,
             );
           },
         ),
-        const SizedBox(width: defaultSpacing),
-        ShowImplementationWidgetsButton(controller: controller),
+        if (controller != null) ...[
+          const SizedBox(width: defaultSpacing),
+          ShowImplementationWidgetsButton(controller: controller!),
+        ],
         const Spacer(),
+        // TODO(https://github.com/flutter/devtools/issues/7860): Clean-up after
+        // Inspector V2 has been released.
+        if (FeatureFlags.inspectorV2) ...[
+          const SizedBox(width: defaultSpacing),
```

Comment:
could this all be in the same spread?

```
if (controller != null && FeatureFlags.inspectorV2) ...[
  const SizedBox(width: defaultSpacing),
  ShowImplementationWidgetsButton(controller: controller!),
  const Spacer(),
  const SizedBox(width: defaultSpacing),
  SwitchSetting(...),
] else 
  const Spacer(),
```

Reply:
Unfortunately it can't! The existence of `controller` means we are on new Inspector, but even for the legacy Inspector we need to show the `SwitchSetting` (so that users can enable the new Inspector)